internal/tui/viewport/tui.go

```
@@ -0,0 +1,337 @@
+)
+type tickMsg time.Duration
+
+// Run a function with a spinner and viewport
+func RunWithSpinner(title string, fn func(chan string, *[]string) (int, error)) (*Model, error) {
+	m := newModel(title, fn)
+	p := tea.NewProgram(m)
+
+	// Conditionally disable input handling when there's no TTY
+	opts := []tea.ProgramOption{}
+	if !m.hasTTY {
+		opts = append(opts, tea.WithInput(strings.NewReader(""))) // 🚀 Prevents unwanted input handling
```

golangci-lint (1.62.2)

ineffectual assignment to opts

(ineffassign)